src/schoology-extractor/tests/test_request_client.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from data_extractor.lib.request_client import RequestClient
+
+pytest.FAKE_KEY = 'TEST_KEY'
+pytest.FAKE_SECRET = 'TEST_SECRET'
+pytest.FAKE_ENDPOINT_URL = 'FAKE_URL'
+pytest.DEFAULT_URL = 'https://api.schoology.com/v1/'
+
+
+@pytest.fixture
+def default_request_client():
+    return RequestClient(pytest.FAKE_KEY, pytest.FAKE_SECRET)
+
+
+class TestRequestClient:
+    class TestInitMethod:
+        def test_default_base_url_is_set(self):
+            request_client = RequestClient(pytest.FAKE_KEY, pytest.FAKE_SECRET)
+            assert request_client.base_url == pytest.DEFAULT_URL
+
+        def test_custom_base_url_is_set(self):
+            custom_url = "a_custom_url"
+            request_client = RequestClient(
+                pytest.FAKE_KEY, pytest.FAKE_SECRET, custom_url)
+            assert request_client.base_url == custom_url
```

Comment:
I like to use a naming convention that is inpsired by Behavior Driven Development, using the words "given, when, then". As you have done here, I usually have nested classes. Typically I switch to "when given then": when structured that way, the action (given) is at a top level, with "when... then" at a more detailed level to encapsulate the variations. What do you think about this?
```suggestion
class TestRequestClient:
    class Test_when_constructing:
        def test_given_not_passing_in_a_url_then_use_the_default_url(self):
            request_client = RequestClient(pytest.FAKE_KEY, pytest.FAKE_SECRET)
            assert request_client.base_url == pytest.DEFAULT_URL

        def test_given_custom_url_then_use_it_for_the_base_url(self):
            custom_url = "a_custom_url"
            request_client = RequestClient(
                pytest.FAKE_KEY, pytest.FAKE_SECRET, custom_url)
            assert request_client.base_url == custom_url
```

Reply:
I will change it